cloud/passportauth/src/user/UserSec.ts

```diff
@@ -0,0 +1,52 @@
+import BaseUser from './BaseUser';
+
+export interface UserSec {
+  getUserId(loginId: string): Promise<string|null>;
+  comparePassword(password: string): Promise<boolean>;
+  hasResourceRole(role: string): Promise<boolean>;
+}
+
+export class UserSecService implements UserSec {
+  protected userApi: BaseUser;
+
+  constructor(protected readonly UserApi: BaseUser) {
+    this.userApi = UserApi;
+  }
+
+  /**
+   * Retrieves the user's id using the user's login id.
+   *
+   * @param loginId {string} - A unique login id
+   * @returns {Promise<string>} - Returns the user's id if found
+   */
+  public getUserId(loginId: string) {
```

Comment:
Why this is needed? IMHO bringing id to this it's a problem. 
We should relay on 3 fields: username, password and role. Username is unique identifier itself. Id is really making our implementation overly complicated.